src/processing.rs

```diff
@@ -79,44 +79,49 @@ pub fn stack_frames(
     );
 
     // Initial necessary values
-    length_signal = sig.shape[0];
-    frame_sample_length = int(np.round(sampling_frequency * frame_length)); // Defined by the number of samples
-    frame_stride = (sampling_frequency * frame_stride).round();
+    let length_signal = sig.shape[0];
+    let frame_sample_length = (sampling_frequency * frame_length).round(); // Defined by the number of samples
+    let frame_stride = (sampling_frequency * frame_stride).round();
+    let mut len_sig = 0;
+    let mut numframes = 0;
 
     // Zero padding is done for allocating space for the last frame.
-    if zero_padding {
+    let signal = if zero_padding {
         // Calculation of number of frames
-        numframes = (int(math.ceil((length_signal - frame_sample_length) / frame_stride)));
-        println!(numframes, length_signal, frame_sample_length, frame_stride);
+        numframes = ((length_signal - frame_sample_length) / frame_stride).ceil() as i32;
+        println!(
+            "{} {} {} {}",
+            numframes, length_signal, frame_sample_length, frame_stride
+        );
 
         // Zero padding
-        len_sig = int(numframes * frame_stride + frame_sample_length);
-        additive_zeros = np.zeros((len_sig - length_signal,));
-        signal = np.concatenate((sig, additive_zeros));
+        len_sig = (numframes * frame_stride + frame_sample_length) as i32;
+        let additive_zeros = ndarray::ArrayBase::zeros((len_sig - length_signal,));
+        ndarray::concatenate(0, &[sig.view(), additive_zeros.view()]);
     } else {
         // No zero padding! The last frame which does not have enough
         // samples(remaining samples <= frame_sample_length), will be dropped!
-        numframes = (length_signal - frame_sample_length).floor() / frame_stride;
+        numframes = (length_signal - frame_sample_length) / frame_stride;
```

Comment:
Hey, there was any reason that the floor was not needed?

Reply:
length_signal and frame_sample_length were both `i32`, and float to int conversion defaults to floor rather than round or ceil, so I didn't believe it was necessary. though I realize they may have just been trying to cast the differance as a float before the division.